ga4gh/datamodel/reads.py

```diff
@@ -68,6 +68,8 @@ class SamFlags(object):
     """
     NUMBER_READS = 0x1
     PROPER_PLACEMENT = 0x2
+    UNMAPPED = 0x4
+    NEXT_SEGMENT_UNMAPPED = 0x8
```

Comment:
Yes, you can distinguish these two states:

`nextMatePosition` gives the mapped position of the next read. null is equivalent to `NEXT_SEGMENT_UNMAPPED`, it does not imply no mate pair.
To determine whether you have a mate you can check `numberReads`, which describes how many reads are in the fragment.

At the bottom of this [migration guide](https://github.com/googlegenomics/readthedocs/blob/master/docs/source/migrating_tips.rst), we have some pseudocode which describes a mapping between SAM flags and GA4GH ReadAlignments.